framework/include/timesteppers/CompositionDT.h

```diff
@@ -38,17 +38,66 @@ class CompositionDT : public TimeStepper
 
   CompositionDT(const InputParameters & parameters);
 
+  struct CompareFirst
+  {
+    bool operator()(const std::pair<Real, TimeStepper *> & a,
+                    const std::pair<Real, TimeStepper *> & b) const
+    {
+      return a.first < b.first;
+    }
+  };
   /**
    * Find the composed time step size by selecting the minimum value and compare it
    * with the lower bound if provided
    * @param dts stores time step size(s) from input time stepper(s)
    * @param bound_dts stores time step size(s) from input lower bound time stepper(s)
    */
-  Real produceCompositionDT(const std::set<Real> & dts, const std::set<Real> & bound_dts);
+  Real produceCompositionDT(std::set<std::pair<Real, TimeStepper *>, CompareFirst> & dts,
+                            std::set<std::pair<Real, TimeStepper *>, CompareFirst> & bound_dts);
 
   // Find the time point to hit at current time step
   Real getSequenceSteppersNextTime();
 
+  template <typename Lambda>
+  void actOnTimeSteppers(Lambda && act);
+
+  /**
+   * Initialize all the input time stepper(s). Called at the very beginning of
+   * Executioner::execute()
+   */
+  virtual void init() override;
```

Comment:
Everything that follows should be override final to denote that nothing should override these later